### PR TITLE
Add MonitoredJobs utility

### LINKED ISSUE #46: Add MonitoredJobs
Depends on #45

### PR DESCRIPTION
**Pull request by chrisrohr**
Closes #46

### CODE REVIEW

File: src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+/**
+ * A set of utilities to assist in setting up MonitoredJobs with health checks.
+ */
+@Slf4j
+@UtilityClass
+public class MonitoredJobs {
+
+    @VisibleForTesting
+    static final Map<String, JobContext> JOBS = newConcurrentMap();
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java:33]: Just use `new ConcurrentHashMap<>()` - there's no real advantage to use Guava for this (and this is exactly and all that `newConcurrentMap()` does)

File: src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                schedule.getIntervalDelay().toSeconds(), TimeUnit.SECONDS);
+    }
+
+    /**
+     * Container that holds the newly created {@link MonitoredJob}, {@link MonitoredJobHealthCheck}
+     * and the {@link Executor} running the job.
+     */
+    @Getter
+    @AllArgsConstructor
+    public static class JobContext {
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java:151]: Who uses this? Actually, how does someone get one? All the above methods return `MonitoredJob`

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java:151]: Hmm... good point. There wasn't anything in the original code that exposed this.  Or did anything with it.

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java:151]: Should I just remove all of that complexity (memory usage)?

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java:151]: It is only used with the concurrent hash map to track if the same job name is used.  I could easily just make that a set.

Comment (chrisrohr) [reply src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java]: That is what I did.

**Review by sleberknight — APPROVED** [review]

**Pull request merged**